src/features/login/ui/LoginForm.tsx

```diff
@@ -1,14 +1,143 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+import AlertBox from "./AlertBox";
+import { useAppDispatch, useAppSelector } from "../../../hooks/hooks";
+import { userLogin } from "../../../slices/loginSlice";
+import { useNavigate } from "react-router-dom";
 
-const LoginForm = () => {
-    return (
-        <div>
-            <form>
-                <input placeholder="Name"/>
-                <button>Submit</button>
-            </form>
-        </div>
-    )
+const schema = yup
+  .object({
+    email: yup
+      .string()
+      .email("Invalid email address")
+      .required("Please enter your email"),
+    password: yup
+      .string()
+      .required("Please enter your password")
+      .min(6, "must be at least 6 characters long"),
+  })
+  .required();
+
+type Inputs = yup.InferType<typeof schema>;
+
+interface User {
+  id: number;
+  name: string;
+  email: string;
+  password: string;
 }
 
-export default LoginForm;
\ No newline at end of file
+const LoginForm = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<Inputs>({
+    resolver: yupResolver(schema),
+  });
+  const [isCorrect, setIsCorrect] = useState<string>("uncheck");
+
+  const { loading, token, error } = useAppSelector((state) => state.login);
+  const dispatch = useAppDispatch();
+
+  const naviagte = useNavigate();
+
+  useEffect(() => {
+    if (token) {
+      console.log("user token in login:", token);
+      naviagte("/");
+    }
+  }, [token, naviagte]);
+
+  const getUser = (email: string, password: string) => {
+    axios
+      .get<User[]>(
+        `http://localhost:3001/Users?email=${email}&password=${password}`
+      )
+      .then((response) => {
+        console.log(response.data);
+        if (response.data.length === 0) {
+          alert("no user found");
+          setIsCorrect("false");
+        } else {
+          alert(response.data);
+          setIsCorrect("true");
+        }
+      });
+  };
```

Comment:
```suggestion
  const getUser = (email: string, password: string) => {
    axios
      .get<User[]>(
        `http://localhost:3001/Users?email=${email}&password=${password}`
      )
      .then((response) => {
        console.log(response.data);
        if (response.data.length === 0) {
          alert("no user found");
          setIsCorrect("false");
        } else {
          alert(response.data);
          setIsCorrect("true");
        }
      });
  };
```
This function is written for testing with JSON server and I think this is no longer needed.